.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,279 @@
+      run: |
+        # NOTE: vars need to be re-assigned
+        # NOTE: set matrix.numpy to pin to a specific numpy version
+        conda_deps="${{ matrix.numpy }} ${MDA_CONDA_${{ matrix.run_type }}_DEPS}"
+        pip_deps=${MDA_PIP_${{ matrix.run_type }}_DEPS}
+        conda install ${conda_deps}
+        pip install ${pip_deps}
+
+        # NOTE: set setuptools version if running asv test
+        if [ ${{ matrix.name }} = "asv_check" ]; then
+          pip install asv "setuptools<45.0.0"
+        fi
+    
+    - name: check_setup
+      run: |
+        # Check OS and python setup
+        echo "OS: ${OS_NAME}"
+        which python
+        which pip
+        pip list
+        conda info
+        conda list
+
+    - name: install_hole
+      if : matrix.install_hole
+      run: |
+        # We manually build hole2 to avoid OS incompatibilities
+        git clone https://github.com/MDAnalysis/hole2.git
+        cd hole2/src
+        source ../source.apache
+        (make FC=${FC}) && (make PREFIX=${HOME}/hole2 FC=${FC} install)
+        source ../source.unset
+        echo "HOLE_BINDIR=${PWD}/../exe" >> $GITHUB_ENV
+        echo "${PWD}/../exe" >> $GITHUB_PATH
+
+    - name: install_mda
+      run: |
+        (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
+
+    - name: run_tests
+      if: contains(matrix.name, 'asv_check') != true
+      run: |
+        PYTEST_FLAGS="--disable-pytest-warnings --durations=50"
+        if [ ${{ matrix.codecov }} = "true" ]; then
+          PYTEST_FLAGS="${PYTEST_FLAGS} --cov=MDAnalysis --cov-report=xml"
+        fi
+        echo $PYTEST_FLAGS
+        pytest -n 2 testsuite/MDAnalysisTests $PYTEST_FLAGS
+
+    - name: run_asv
+      if: contains(matrix.name, 'asv_check')
+      run: |
+        cd benchmarks
+        time python -m asv check -E existing
+
+    - name: codecov
+      if: matrix.codecov
+      uses: codecov/codecov-action@v1
+      with:
+        file: coverage.xml
+        fail_ci_if_error: True
+        verbose: True
+
+
+  build_docs:
+    if: "github.repository == 'MDAnalysis/mdanalysis'"
+    runs-on: ubuntu-latest
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: 3.7
+        auto-update-conda: true
+        channel-priority: flexible
+        channels: biobuilds, conda-forge
+        add-pip-as-python-dependency: true
+        architecture: x64
+
+    - name: install_deps
+      run: |
+        conda_deps="${{ env.MDA_CONDA_MIN_DEPS }} ${{ env.MDA_CONDA_EXTRA_DEPS}}"
+        pip_deps="${{ env.MDA_PIP_MIN_DEPS}} ${{ env.MDA_PIP_EXTRA_DEPS }} sphinx==1.8.5 sphinx-sitemap sphinx_rtd_theme msmb_theme==1.2.0"
+        conda install ${conda_deps}
+        pip install ${pip_deps}
+
+    - name: install_mda
+      run: |
+        cd package && python setup.py build_ext --inplace
```

Comment:
Why --inplace instead of a normal installation?

Reply:
Honestly no idea, it's one of those things I just copied over from the initial travis yaml file. I'll see if we can live without it.

Reply:
So it looks like it relates back to: https://github.com/MDAnalysis/mdanalysis/commit/80bc2a2a9cc826627eca2691e425c284fb76269f and https://github.com/MDAnalysis/mdanalysis/issues/386#issuecomment-139747327. I've tried iterating through a couple of different options, if we switch to `python setup.py build_sphinx -E` in the next step (instead of `python setup.py build_sphinx`), then we can use `python setup.py develop` here (note that `install` won't work because of cython [issues](https://github.com/IAlibay/mdanalysis/runs/1484029180?check_suite_focus=true)).

I have changed it to using python setup.py develop` since it's more in line with all the other things we do. The only thing I'm not sure about is if/how this would affect docs deployment.